packages/grafana-data/src/types/datasource.ts

```diff
@@ -231,6 +240,13 @@ abstract class DataSourceApi<
    */
   abstract query(request: DataQueryRequest<TQuery>): Promise<DataQueryResponse> | Observable<DataQueryResponse>;
 
+  /**
+   * For a given query return a set of related data providers - object that can return data related to the
+   * given query, for example: logs volume for logs query.
+   * @param request
+   */
+  getQueryRelatedDataProviders?(request?: DataQueryRequest<TQuery>): QueryRelatedDataProviders;
```

Comment:
I would probably keep it beta/alpha/internal or something until we support other usecases than the explore logs. It will also give us some more space to break it if needed for few releases I guess.

Reply:
Added in db9fb89c37a4e79c599073845c0c3359131b58b2